cs3api4lab/exception/exceptions.py

```diff
@@ -1,77 +1,87 @@
-class ShareError(Exception):
-    def __init__(self, message):
-        self.message = message
-        super().__init__(self.message)
-
-    def __str__(self):
-        return self.__class__.__name__ + ": " + self.message
-
-class ShareAlreadyExistsError(Exception):
-    def __init__(self, message):
-        self.message = message
-        super().__init__(self.message)
-
-    def __str__(self):
-        return self.__class__.__name__ + ": " + self.message
-
-
-class ShareNotFoundError(Exception):
-    def __init__(self, message):
-        self.message = message
-        super().__init__(self.message)
-
-    def __str__(self):
-        return self.__class__.__name__ + ": " + self.message
-
-
-class ResourceNotFoundError(Exception):
-    def __init__(self, message):
-        self.message = message
-        super().__init__(self.message)
-
-    def __str__(self):
-        return self.__class__.__name__ + ": " + self.message
-
-
-class ProviderNotFoundError(Exception):
-    def __init__(self, message):
-        self.message = message
-        super().__init__(self.message)
-
-    def __str__(self):
-        return self.__class__.__name__ + ": " + self.message
-
-class LockNotFoundError(Exception):
-    def __init__(self, message):
-        self.message = message
-        super().__init__(self.message)
-
-    def __str__(self):
-        return self.__class__.__name__ + ": " + self.message
-
-
-class OCMDisabledError(Exception):
-    def __init__(self, message):
-        self.message = message
-        super().__init__(self.message)
-
-    def __str__(self):
-        return self.__class__.__name__ + ": " + self.message
-
-
-class InvalidTypeError(Exception):
-    def __init__(self, message):
-        self.message = message
-        super().__init__(self.message)
-
-    def __str__(self):
-        return self.__class__.__name__ + ": " + self.message
-
-
-class ParamError(Exception):
-    def __init__(self, key_error):
-        self.message = "Missing argument: " + str(key_error)
-        super().__init__(self.message)
-
-    def __str__(self):
+class ShareError(Exception):
+    def __init__(self, message):
+        self.message = message
+        super().__init__(self.message)
+
+    def __str__(self):

```

Comment:
I'm not sure why this appears as changed, but can you create a new PR where you fix these exceptions?

Instead of repeating the `__str__` on all of them, just create a new class, that inherits from `Exception`, and use that one as the parent of all the others.

Reply:
For later, ofc

Reply:
noted